### ISSUE TITLE
Library Function - Same cart is randomly repeated multiple times in display list

### ISSUE THREAD

**Issue by BeDazzleMe** [olsson82/rivendellweb#22]
Running on ubuntu with RD 4.3.
When I click into Library and move through the list of carts available, many carts are repeated in the list, often 3 to 5 times before moving onto the next cart.

It appears to be random and does not happen for every cart.

I have attached a screen shot of what I see.

![rdweb-library-carts-repeated](https://github.com/user-attachments/assets/e1ecc659-8eac-428a-84f7-df4a6f0c0e09)

**Comment by olsson82** [comment]
Strange need to investigate this. Is this when you have all groups and all scheduler codes activated?

Does it do the same if you select one group or seduler code also ?

**Comment by BeDazzleMe** [comment]
Hi Andreas,
Currently testing to work out any patterns or permissions issues.
My RD 4.3 system runs in test/staging and is a restored 3 system which is updated daily.
Let me double check it's not related to the restore and permissions not appearing.
I will come back to you.

**Comment by BeDazzleMe** [comment]
Hi Andreas,
I have been able to replicate this issue and found that the repeat carts are tied to scheduler codes.
If there are 5 assigned scheduler codes, the cart will appear 5 times in the library list.
To confirm this, I removed 1 scheduler code for a cart and refreshed the library list which showed 1 less cart.
I then removed all but 1 scheduler code and the library list now shows only 1 cart.
Hope this helps.
Darren.
